src/main/kotlin/com/terraformation/backend/db/IdentifierGenerator.kt

```diff
@@ -1,65 +1,86 @@
-   * that starts at 000 for the first identifier created on a particular date. The desired behavior
-   * is for the suffix to represent the order in which entries were added to the system, so ideally
-   * we want to avoid gaps or out-of-order values, though it's fine for that to be best-effort.
-   *
-   * The implementation uses a database sequence. The sequence's values follow the same pattern as
-   * the identifiers, but the suffix is always 10 digits; it is rendered as a 3-or-more-digit value
-   * by this method.
+   * Generates a new identifier for an organization.
    *
-   * If the date part of the sequence value doesn't match the current date, this method resets the
-   * sequence to the zero suffix for the current date.
-   *
-   * Note that there is a bit of a race condition if multiple terraware-server instances happen to
-   * allocate their first identifier of a given day at the same time; they might both reset the
-   * sequence. To guard against that, [AccessionStore.create] will retry a few times if it gets a
-   * unique constraint violation on the accession number.
+   * @param timeZone Time zone to use to determine the current date.
    */
-  fun generateIdentifier(): String {
+  fun generateIdentifier(
+      organizationId: OrganizationId,
+      timeZone: ZoneId = ZoneOffset.UTC,
+  ): String {
     val suffixMultiplier = 10000000000L
-    val todayAsLong = LocalDate.now(clock).format(DateTimeFormatter.BASIC_ISO_DATE).toLong()
+    val todayAsLong =
+        LocalDate.ofInstant(clock.instant(), timeZone)
+            .format(DateTimeFormatter.BASIC_ISO_DATE) // "20221031"
+            .toLong()
+    val firstValueForToday = todayAsLong * suffixMultiplier
 
     val sequenceValue =
-        dslContext.select(ACCESSION_NUMBER_SEQ.nextval()).fetchOne(ACCESSION_NUMBER_SEQ.nextval())!!
-    val datePart = sequenceValue / suffixMultiplier
-    val suffixPart = sequenceValue.rem(suffixMultiplier)
-
-    val suffix =
-        if (todayAsLong != datePart) {
-          val firstValueForToday = todayAsLong * suffixMultiplier
+        with(IDENTIFIER_SEQUENCES) {
           dslContext
-              .alterSequence(ACCESSION_NUMBER_SEQ)
-              .restartWith(firstValueForToday + 1)
-              .execute()
-          log.info("Resetting identifier sequence to $firstValueForToday")
-          0
-        } else {
-          suffixPart
+              .insertInto(IDENTIFIER_SEQUENCES)
+              .set(ORGANIZATION_ID, organizationId)
+              .set(NEXT_VALUE, firstValueForToday)
+              .onDuplicateKeyUpdate()
+              .set(NEXT_VALUE, DSL.greatest(NEXT_VALUE.plus(1), DSL.value(firstValueForToday)))
+              .returning(NEXT_VALUE)
+              .fetchOne(NEXT_VALUE)!!
         }
 
-    return "%08d%03d".format(todayAsLong, suffix)
+    val datePart = sequenceValue / suffixMultiplier
+    val suffixPart = sequenceValue.rem(suffixMultiplier)
+
+    return "%08d%03d".format(datePart, suffixPart)
```

Comment:
oh, gotcha, `%03d` does not do a mod 1000, it just makes sure to pad with zeros out to _at least_ 3 digits (had to go look that up 😅 )